internal/ethapi/api.go

```diff
@@ -354,19 +354,19 @@ func (s *PersonalAccountAPI) DeriveAccount(url string, path string, pin *bool) (
 }
 
 // NewAccount will create a new account and returns the address for the new account.
-func (s *PersonalAccountAPI) NewAccount(password string) (common.Address, error) {
+func (s *PersonalAccountAPI) NewAccount(password string) (string, error) {
```

Comment:
An alternative approach is to define the customized json marshaller

e.g. 

```go
func (a Address) MarshalJSON() ([]byte, error) {
	return json.Marshal(a.Hex())
}
```

But in this case, the address will always be marshalled in checksum'd format. I am not sure this is what you want.

Reply:
Or we can define `ChecksumedAddress` type which has this customized marshal rule and return it instead.

Reply:
Like adding something like this in [types.go](https://github.com/ethereum/go-ethereum/blob/master/common/types.go)?

```go
type AddressEIP55 [AddressEIP55Length]byte // AddressEIP55Length is 42

// NewAddressEIP55 creates a new AddressEIP55 from an Address.
func NewAddressEIP55(addr Address) AddressEIP55 {
	return BytesToAddressEIP(addr.checksumHex())
}

func BytesToAddressEIP(b []byte) AddressEIP55 {
	var a AddressEIP55
	copy(a[:], b)
	return a
}

func (a AddressEIP55) String() string {
	return string(a[:])
}
```

Reply:
```diff
diff --git a/common/types.go b/common/types.go
index 218ca0be4c..1431857372 100644
--- a/common/types.go
+++ b/common/types.go
@@ -429,3 +429,12 @@ func (ma *MixedcaseAddress) ValidChecksum() bool {
 func (ma *MixedcaseAddress) Original() string {
 	return ma.original
 }
+
+// AddressEIP55 is an alias of Address with a customized json marshaller
+// defined in the manner of EIP55.
+type AddressEIP55 Address
+
+// MarshalJSON marshals the address in the manner of EIP55.
+func (addr AddressEIP55) MarshalJSON() ([]byte, error) {
+	return json.Marshal(Address(addr).Hex())
+}
diff --git a/common/types_test.go b/common/types_test.go
index 94492278d8..d41b4b1d7e 100644
--- a/common/types_test.go
+++ b/common/types_test.go
@@ -534,3 +534,21 @@ func TestHash_Format(t *testing.T) {
 		})
 	}
 }
+
+func TestAddressEIP55(t *testing.T) {
+	addr := HexToAddress("0x5aaeb6053f3e94c9b9a09f33669435e7ef1beaed")
+	blob, err := json.Marshal(AddressEIP55(addr))
+	if err != nil {
+		t.Fatal("Failed to marshal AddressEIP55", err)
+	}
+	if strings.Trim(string(blob), "\"") != addr.Hex() {
+		t.Fatal("Address with checksum is expected")
+	}
+	var dec Address
+	if err := json.Unmarshal(blob, &dec); err != nil {
+		t.Fatal("Failed to unmarshal AddressEIP55", err)
+	}
+	if addr != dec {
+		t.Fatal("Unexpected address after unmarshal")
+	}
+}

```

A very quick implementation. You can probably polish a bit on top.

Reply:
Thanks for the feedback @rjl493456442! 

Pushed a commit that addresses this properly I hope!